azure/datalake/store/lib.py

```diff
@@ -86,6 +90,8 @@ def auth(tenant_id=default_tenant, username=default_username,
         the secret associated with the client_id
     resource : str
         resource for auth (e.g., https://management.core.windows.net/)
+    require_2fa : bool
+        indicates this authentication attempt requires two-factor authentication
```

Comment:
@lmazuel for now I am ok with us continuing with what we have and updating the internal logic later when we have a standardized pattern that I can use.